src/main/java/baseball/Application.java

```diff
@@ -1,7 +1,187 @@
 package baseball;
 
-public class Application {
+import camp.nextstep.edu.missionutils.Console;
+import camp.nextstep.edu.missionutils.Randoms;
+
+public class Application implements BaseballGame {
+    final String CMD_NEW_GAME = "1";
+    final String CMD_END_GAME = "2";
+    final int AMOUNT_OF_NUMBERS = 3;
+    final int NUMBER_RANGE_START_INCLUSIVE = 1;
+    final int NUMBER_RANGE_END_INCLUSIVE = 9;
+    int[] hostNumbers = new int[] { -1, -1, -1 };
+    int[] playerNumbers = new int[] { -1, -1, -1 };
+
     public static void main(String[] args) {
-        //TODO: 숫자 야구 게임 구현
+        Application app = new Application();
+        do {
+            app.start();
+        } while (app.promptStartNewGame());
```

Comment:
여기서 `app.start()`로 실행했지만 `end`메서드는 따로 보이지 않는군요. 알고보니 `start`내부에 `end`도 같이 있는 구조네요. 우리가 생각하는 영문에서의 의미는 `start`라는 단어와 `end`라는 단어가 같은 레벨에서 받아들여집니다. 다시말해, 내가 보고 있는 어딘가에서 `start`가 되었다면, 같은 수준(여기서는 메서드 위치)에서 `end`또한 등장할거라 기대할 수 있다는 의미이죠. 지금의 구조에서는 `start`내부에 `end`가 있으니 약간 어색합니다.

구조 자체는 잘 잡혀 있으므로 이름만 바꿔보는건 어떨까요? 사실 지금의 `start`메서드는 *게임 시작*만 하는게 아니라 게임의 모든 과정을 총괄하는 역할이니까요!

Reply:
선생님 말씀을 듣고보니 아래의 표현이 조금 더 일반적으로 기대대는 수준에 맞는 레벨이 되겠네요!

```suggestion
            app.start();
            app.end();
        } while (app.promptStartNewGame());
```

Reply:
`start()` 메소드 속에서 호출되던 `end()`를 밖에서 실행하게 되었으니, 이에 맞게 `start()` 함수도 변경해야겠군요. 수정해보겠습니다 :)

Reply:
[80333fa](https://github.com/kth496/java-baseball-implement/pull/5/commits/80333fa1b5cc2c847e77f89e02cda6a351d3d766) 에서 `start()`를 `run()`으로 rename하고, 구조를 개편함으로서 해결했습니다.